generate.py

```diff
@@ -229,7 +229,7 @@ def _load_model(checkpoint_path, device, precision, use_tp):
         assert path_comps[-2].startswith("g")
         groupsize = int(path_comps[-2][1:])
         from quantize import WeightOnlyInt4QuantHandler
-        simple_quantizer = WeightOnlyInt4QuantHandler(model, groupsize)
+        simple_quantizer = WeightOnlyInt4QuantHandler(model, groupsize, device=device)
```

Comment:
I have 2 questions here: 
1 - do we actually need the device?  Usually the device is the one that inputs are on?  
2 - we later base decisions on what seems to be block size and dtype on how to represent this.  If we get Weights that were quantized on CPU, will they work on CUDA, and vice versa? (This has historically been the goal for our operators) - add'l comments below.